src/dev/flang/be/jvm/runtime/Intrinsics.java

```diff
@@ -811,6 +811,21 @@ public static boolean fuzion_sys_env_vars_has0(Object s)
     return System.getenv(Runtime.utf8ByteArrayDataToString((byte[])s)) != null;
   }
 
+  public static void fuzion_sys_thread_join0(long threadId)
+  {
+    try
+      {
+        Runtime._startedThreads_.get(threadId).join();
+        Runtime._startedThreads_.remove(threadId);
+      }
+    catch (InterruptedException e)
+      {
+        // NYI handle this exception
+        System.err.println("Joining of threads was interrupted: " + e);
+        System.exit(1);
+      }
+  };
```

Comment:
Better put this into a loop that repeats `join` until it finished without exception. 
BTW, this code is not thread-safe: ` _startedThread_.get()`  could result in `null` causing a `NullPointerException`. Furthermore, ` remove` should probably not be called by `join`, but either by the Thread itself or by some cleanup mechanism that removes terminated threads, either when new threads are started or by a system thread that joins and removes threads that are about to terminate.

Reply:
@fridis ~~My understanding is that this does not need to be thread-safe since fuzion stdlib only lets you join a thread that the thread calling `join` spawned.~~ 

Seems like this is not the case (yet). But probably should be.

My question is:
Are there any use-cases where you want to join a thread from a different thread than the one that started it?

Also there is this comment on `join` in stdlib:
# NYI it should be made impossible to call this more than once

Reply:
from `man pthread`:
> If multiple threads simultaneously try to join with the same thread, the results are undefined

Reply:
I was not aware of this. In Java. `Thread.join()` can be called from any thread to any other thread and an arbitrary number of times. Should IMHO be the same in our base lib, so we eventually will need to have a more sophisticated handling here. For now, since we do not have any mechanism like Java's `wait`/`notify`, I should be enough to document that his needs more work.

Reply:
I added issue #2161